patches/aws-lc/0009-Adding-detection-of-out-of-bound-pre-bound-memory-re.patch

```diff
@@ -0,0 +1,504 @@
+From 9c21a0cdcddc69130ea9a6e116f2a9d9e2422cc1 Mon Sep 17 00:00:00 2001
+From: Nevine Ebeid <66388554+nebeid@users.noreply.github.com>
+Date: Fri, 28 Mar 2025 12:13:41 -0400
+Subject: [PATCH] Adding detection of out-of-bound pre-bound memory read to
+ AES-XTS tests. (#2286)
+
+This change hardens the tests introduced in #2227 Fix out-of-bound (OOB)
+input read in AES-XTS Decrypt in AVX-512 implementation.
+It adds a memory page preceding the input and output buffer that is
+protected against read and write in order to detect any under-read, in
+which case a segfault occurs.
+
+The suspected code that can potentially cause a "pre-bound" OOB is the
+cipher-stealing section in Encrypt
+
+[crypto/fipsmodule/aes/asm/aesni-xts-avx512.pl#L1809-L1810](https://github.com/aws/aws-lc/blob/v1.48.5/crypto/fipsmodule/aes/asm/aesni-xts-avx512.pl#L1809-L1810)
+and decrypt
+
+[crypto/fipsmodule/aes/asm/aesni-xts-avx512.pl#2572-L2573](https://github.com/aws/aws-lc/blob/v1.48.5/crypto/fipsmodule/aes/asm/aesni-xts-avx512.pl#L2572-L2573).
+
+The efficacy of the added test was shown by changing the decrypt
+cipher-stealing code for example to:
+# ```diff
+# --- a/crypto/fipsmodule/aes/asm/aesni-xts-avx512.pl
+# +++ b/crypto/fipsmodule/aes/asm/aesni-xts-avx512.pl
+# @@ -2569,7 +2569,7 @@ ___
+#    vpshufb       %xmm10,%xmm8,%xmm8
+
+# -  vmovdqu       -0x10($input,$length,1),%xmm3
+# +  vmovdqu       -0x12($input,$length,1),%xmm3
+#    vmovdqu       %xmm8,-0x10($output,$length,1)
+#  ```
```

Comment:
I modified this diff block with comment prefix (`#`) because it is a diff block in the commit message, not actual diff block to be applied. It would confuse `git` if not make them as comments.